copy/multiple.go

```diff
@@ -27,23 +32,90 @@ const (
 type instanceCopy struct {
 	op           instanceCopyKind
 	sourceDigest digest.Digest
+
+	// Fields which can be used by callers when operation
+	// is `instanceCopyClone`
+	compressionVariant            *compression.Algorithm
+	cloneAddPlatform              *imgspecv1.Platform
+	cloneAddAnnotations           map[string]string
+	cloneAddCompressionAlgorithms []compression.Algorithm
+}
+
+// internal type only to make imgspecv1.Platform comparable
+type platformComparable struct {
+	Architecture string
+	OS           string
+	OSVersion    string
+	OSFeatures   string
+	Variant      string
+}
+
+// Converts imgspecv1.Platform to a comparable format.
+func platformV1ToPlatformComparable(platform imgspecv1.Platform) platformComparable {
+	return platformComparable{Architecture: platform.Architecture,
+		OS:         platform.OS,
+		OSFeatures: strings.Join(platform.OSFeatures, ","),
```

Comment:
This is not _quite_ correct, nothing prevents elements of `OSFeatures` from including `,`. Probably good enough in practice, maybe just worth a comment.

---

For more correct matching, `strings.Join(sort.Strings(platform.OSFeatures))` would help.